tests/containers/bci_repo.pm

```diff
@@ -0,0 +1,63 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: bci-repo test
+#   SUSE Linux Enterprise Base Container Images (SLE BCI)
+#   provides truly open, flexible and secure container images and application
+#   development tools for immediate use by developers and integrators without
+#   the lock-in imposed by alternative offerings.
+#
+#   This module is used to test BCI repository.
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+
+sub container_exec {
+    my $container = shift;
+    my $cmd = shift;
+    assert_script_run(qq[podman exec $container /bin/sh -c '$cmd'], @_);
+}
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $image = get_required_var('CONTAINER_IMAGE_TO_TEST');
+    my $bci_repo = get_required_var('BCI_REPO_TO_TEST');
+    my $container = 'bci-repo-tester';
+    my $patterns = join(" ", split(/\s*,\s*/, get_required_var("BCI_REPO_PATTERNS")));
+
+    # download and start a BCI container
+    record_info('IMAGE', $image);
+    script_retry("podman pull $image", timeout => 300, delay => 60, retry => 3);
+    record_info('Inspect', script_output("podman inspect $image"));
+    assert_script_run("podman run --name $container -dt $image");
+
+    # query default setup
+    container_exec($container, 'zypper ref', timeout => 180);
+    container_exec($container, 'zypper lr -d');
+    container_exec($container, 'zypper search -t pattern', timeout => 180);
+    container_exec($container, "zypper -q -s 11 pa --orphaned | tee -a repo.org", timeout => 600);
+    # remove container-suseconnect, in order to install the Cloud patterns
+    container_exec($container, 'zypper -n rm container-suseconnect', timeout => 180);
+    # remove SLE_BCI repo pointing to official update servers and add SUT repo
+    container_exec($container, 'zypper rr 1');
+    container_exec($container, "zypper ar $bci_repo BCI_TEST");
+    container_exec($container, 'zypper ref', timeout => 180);
+    container_exec($container, 'zypper lr -d');
+    container_exec($container, 'zypper search -t pattern', timeout => 180);
+    container_exec($container, "zypper -n in -t pattern $patterns", timeout => 600);
```

Comment:
This caused some issues to me, I couldn't install all patterns at once... There were conflicting packages for the different public cloud provider patterns, so I needed to install them separately. Maybe a loop for each pattern and installing it in fresh container?

Reply:
It should install patterns according to specified variable as it is mentioned in the description of the ticked. Thus in this case it has pulled the group with `Google_Cloud_Platform,Google_Cloud_Platform_Instance_Init,Google_Cloud_Platform_Instance_Tools,Google_Cloud_Platform_Tools`.

Reply:
Yes, and I probably forgot to update the ticket with the outcome of my manual testing... 
btw, the zypper command should return failure if it fails, in the [VR](http://kepler.suse.cz/tests/16450#step/bci_repo/55) I see
`System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down`